internal/injector/aspect/join/struct.go

```diff
@@ -78,6 +79,30 @@ func (s *structLiteral) Matches(chain *node.Chain) bool {
 		return s.matchesLiteral(chain)
 	}
 
+	if strings.HasPrefix(s.field, "-") {
+		if !s.matchesLiteral(chain) {
+			return false
+		}
+
+		// If `matchesLiteral` is true, this is guaranteed to be a *dst.CompositeLit
+		lit := chain.Node.(*dst.CompositeLit)
+		for _, elt := range lit.Elts {
+			kve, ok := elt.(*dst.KeyValueExpr)
+			if !ok {
+				continue
+			}
+			key, ok := kve.Key.(*dst.Ident)
+			if !ok {
+				continue
+			}
+			if key.Name == s.field[1:] {
+				return false
+			}
+		}
+
+		return true
+	}
+
```

Comment:
I also don't see this code used anywhere on the net/http client yml file. Am I missing something?